temporal-sdk/src/main/java/io/temporal/common/converter/SearchAttributesUtil.java

```diff
@@ -0,0 +1,124 @@
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.common.converter;
+
+import io.temporal.api.common.v1.Payload;
+import io.temporal.api.common.v1.SearchAttributes;
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nonnull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SearchAttributesUtil {
+  private static final SearchAttributesPayloadConverter converter =
+      SearchAttributesPayloadConverter.INSTANCE;
+  private static final Logger log = LoggerFactory.getLogger(SearchAttributesUtil.class);
+
+  public static SearchAttributes encode(Map<String, Object> searchAttributes) {
+    return SearchAttributes.newBuilder()
+        .putAllIndexedFields(objectToPayloadMap(searchAttributes))
+        .build();
+  }
+
+  public static Map<String, Object> decode(SearchAttributes searchAttributes) {
+    if (searchAttributes == null || searchAttributes.getIndexedFieldsCount() == 0) {
+      return null;
+    }
+    return payloadToObjectMap(searchAttributes.getIndexedFieldsMap());
+  }
+
+  public static Map<String, Payload> objectToPayloadMap(Map<String, Object> map) {
+    if (map == null || map.isEmpty()) {
+      return null;
+    }
+    Map<String, Payload> payloadMap = new HashMap<>(map.size());
+    map.forEach((key, value) -> payloadMap.put(key, converter.toData(value).get()));
+    return payloadMap;
+  }
+
+  public static Map<String, Object> payloadToObjectMap(Map<String, Payload> serializedMap) {
+    if (serializedMap == null || serializedMap.isEmpty()) {
+      return null;
+    }
+    Map<String, Object> deserializedMap = new HashMap<>();
+    for (Map.Entry<String, Payload> attribute : serializedMap.entrySet()) {
+      String key = attribute.getKey();
+      Object data = converter.fromData(attribute.getValue());
+      if (data == null) {
+        log.error("Error parsing Search Attribute: {}.", key);
+      } else {
+        deserializedMap.put(key, data);
+      }
+    }
+    return deserializedMap;
+  }
+
+  @Nonnull
+  public static SearchAttributeType javaTypeToEncodedType(Class<?> type) {
+    if (String.class.equals(type)) {
+      return SearchAttributeType.String;
+    } else if (Integer.class.equals(type) || Short.class.equals(type) || Byte.class.equals(type)) {
+      return SearchAttributeType.Int;
+    } else if (Double.class.equals(type) || Float.class.equals(type)) {
+      return SearchAttributeType.Double;
+    } else if (Boolean.class.equals(type)) {
+      return SearchAttributeType.Bool;
+    } else if (LocalDateTime.class.equals(type)) {
+      return SearchAttributeType.Datetime;
+    }
+    return SearchAttributeType.Unspecified;
+  }
+
+  public enum RegisteredSearchAttributes {
```

Comment:
`Registered` is misleading here. Registered are the ones that users registered.
We actually use the term "System" and I would stick to what is already in our documentation: https://docs.temporal.io/docs/server/workflow-search/#system-attributes

Reply:
This enum is and should be legitimately public.
 It should be moved from "io.temporal.common.converter" into `io.temporal.workflow` and be a first-class citizen, not an inner enum.